### PR TITLE
ACS-1630 : Storage classes - REST api - GET /nodes/{nodeId}

### PR DESCRIPTION
**Pull request by tzclucian**

### CODE REVIEW

File: remote-api/src/main/java/org/alfresco/rest/api/impl/NodesImpl.java [hunk 1]
```diff
@@ -1048,6 +1048,9 @@ else if (permissionService.hasPermission(nodeRef, perm) == AccessStatus.ALLOWED)
         node.setNodeType(nodeTypeQName.toPrefixString(namespaceService));
         node.setPath(pathInfo);
 
+
+        node.getContent().setStorageClasses(contentService.findStorageClasses(nodeRef));
```
Comment (cristianturlica) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/NodesImpl.java:1052]: Please add check `if (includeParam.contains(PARAM_INCLUDE_STORAGE_CLASSES))` before requesting the storage classes info as per specs.... This action is performance intense and should not be done without request.

File: remote-api/src/main/java/org/alfresco/rest/framework/resource/content/BinaryProperty.java [hunk 2]
```diff
@@ -73,6 +73,7 @@ public BinaryProperty(ContentReader reader)
         this.encoding = reader.getEncoding();
         this.length = reader.getSize();
         this.locale = reader.getLocale();
+        this.storageClasses = Collections.emptySet();
```
Comment (cristianturlica) [inline remote-api/src/main/java/org/alfresco/rest/framework/resource/content/BinaryProperty.java:76]: Is this ok to an empty set? How do we distinguish when we set the value from when we do not?

**Review by cristianturlica — CHANGES_REQUESTED** [review]

File: remote-api/src/main/java/org/alfresco/rest/api/impl/NodesImpl.java (hunk 1 above)
Comment (tzclucian) [reply remote-api/src/main/java/org/alfresco/rest/api/impl/NodesImpl.java]: Done.

File: remote-api/src/main/java/org/alfresco/rest/framework/resource/content/BinaryProperty.java (hunk 2 above)
Comment (tzclucian) [reply remote-api/src/main/java/org/alfresco/rest/framework/resource/content/BinaryProperty.java:76]: Updated.

**Comment by tzclucian** [comment]
Ready for review.

File: remote-api/src/main/java/org/alfresco/rest/api/model/ContentInfo.java [hunk 3]
```diff
@@ -70,10 +82,20 @@ public String getEncoding() {
...
+	{
+		this.storageClasses = storageClasses;
+	}
+
+	@Override
 	public String toString()
 	{
 		return "ContentInfo [mimeType=" + mimeType + ", mimeTypeName=" + mimeTypeName
-				+ ", encoding=" + encoding + ", sizeInBytes=" + sizeInBytes + "]";
+			+ ", encoding=" + encoding + ", sizeInBytes=" + sizeInBytes + ",storageClasses=" + storageClasses + "]";
```
Comment (ancutam) [inline remote-api/src/main/java/org/alfresco/rest/api/model/ContentInfo.java:99]: A space before `storageClasses`.

File: remote-api/src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java [hunk 4]
```diff
@@ -4778,6 +4779,37 @@ private void testRetrieveNodePermissions() throws Exception
...
+
+        Map params = new HashMap<>();
+        params.put("include", "storageClasses");
+
+        // Update node
+        Document dUpdate = new Document();
+        HttpResponse response = put(URL_NODES, docId, toJsonAsStringNonNull(dUpdate), null, 200);
+        Document documentResp = RestApiUtil.parseRestApiEntry(response.getJsonResponse(), Document.class);
+
+        // Check if storageClasses are retrieved if 'include=permissions' is not sent in the request
```
Comment (ancutam) [inline remote-api/src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java:4799]: `include=storageClasses`

**Review by ancutam — APPROVED** [review]
Looks good, just two super minor changes.

File: remote-api/src/main/java/org/alfresco/rest/api/model/ContentInfo.java (hunk 3 above)
Comment (tzclucian) [reply remote-api/src/main/java/org/alfresco/rest/api/model/ContentInfo.java]: Thank you, totally missed it.

File: remote-api/src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java (hunk 4 above)
Comment (tzclucian) [reply remote-api/src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java]: Thank you, updated.

**Comment by tzclucian** [comment]
@cristianturlica , can you please recheck the PR and update your review on it?

**Review by cristianturlica — COMMENTED** [review]
Changes look good to me.

**Pull request merged**